src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java

```diff
@@ -106,16 +109,24 @@ public CassandraClient execute(String query, Handler<AsyncResult<ResultSet>> res
   @Override
   public CassandraClient execute(Statement statement, Handler<AsyncResult<ResultSet>> resultHandler) {
     executeWithSession(session -> {
-      ResultSetFuture resultSetFuture = session.executeAsync(statement);
-      Future<com.datastax.driver.core.ResultSet> vertxExecuteFuture = Util.toVertxFuture(resultSetFuture, vertx);
-      vertxExecuteFuture.setHandler(executionResult -> {
-        if (executionResult.succeeded()) {
+      Future<com.datastax.driver.core.ResultSet> future = Util.toVertxFuture(session.executeAsync(statement), vertx);
+      List<Row> resultRows = new LinkedList<>();
```

Comment:
Why should we use `ArrayList` here? I think that if the result set is big, it can cause thread blocks when `ArrayList` will be resizing.

In fact, for now, for accessing `ResultSet` user can do it with the `ResultSet#iterator()` call, so i think `LinkedList` is both fines for adding new elements when fetching and for accessing with `iterator`.

Reply:
We also can use combined approach: using `LinkedList` when adding a lot of element while fetching and copy all the elements to the `ArrayList` when we are done.

Reply:
If we block the event loop because the ArrayList is too big then the user really shouldn't be getting all the results at once.

Seriously, this won't happen in practice. LinkedList should be avoided.

Reply:
Yes, I agree, let's use an `ArrayList` here.

Reply:
Done